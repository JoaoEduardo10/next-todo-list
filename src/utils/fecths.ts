import { json } from 'stream/consumers';

export type TDataProps = {
  email: string;
  password: string;
};
interface IUser {
  name: string;
  email: string;
  password: string;
}
const urlApi = process.env.NEXT_PUBLIC_URL_API as string;

const loginUser = async (data: TDataProps) => {
  const response = await fetch(`${urlApi}/login`, {
    method: 'POST',
    headers: {
      'content-type': 'application/json',
    },
    body: JSON.stringify(data),
  });

  const token = await response.json();

  return token;
};

const createUser = async (user: IUser) => {
  const { email, name, password } = user;

  const response = await fetch(`${urlApi}/users`, {
    method: 'POST',
    headers: {
      'content-type': 'application/json',
    },
    body: JSON.stringify({ email, password, name }),
  });

  return {
    ok: response.ok,
  };
};

export { loginUser, createUser };
