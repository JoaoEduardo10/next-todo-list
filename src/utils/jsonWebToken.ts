import jwt from 'jsonwebtoken';

export interface Ijwt {
  email: string;
  name: string;
  id: string | number;
}

export interface IjwtComplete {
  iat: number;
  exp: number;
}

export const compareJwt = (token: string): (Ijwt & IjwtComplete) | null => {
  try {
    const jwtHash = process.env.NEXT_PUBLIC_HASH_JWT as string;

    const varifyToken = jwt.verify(token, jwtHash);

    return varifyToken as Ijwt & IjwtComplete;
  } catch (error: any) {
    console.log(error.message);
    return null;
  }
};
