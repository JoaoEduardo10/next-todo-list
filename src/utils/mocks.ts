import configureStore from 'redux-mock-store';
import { TBoard } from '../types';
import { mockTask } from '../components/task/mock';

const mockStore = configureStore([]);

const mockBoards: TBoard[] = [
  { id: 'board1', boardName: 'Board 1', taskConnect: '123', userId: '2' },
  { id: 'board2', boardName: 'Board 2', taskConnect: '1234', userId: '2' },
];

const store = mockStore({
  boards: {
    allBoards: [
      { id: 'board1', boardName: 'Board 1' },
      { id: 'board2', boardName: 'Board 2' },
    ],
    actualBoard: { id: 'board1', boardName: 'Board 1' },
    actualBoardWithTasks: {
      boardName: 'Board 1',
      id: '1234',
      taskConnect: '1234',
      tasks: mockTask,
      userId: '1234',
    },
  },
});

export { store, mockBoards };
