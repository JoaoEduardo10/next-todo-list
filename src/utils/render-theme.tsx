import { render, RenderResult } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { theme } from '../styles/theme';
import { store as storeActual } from '../app/store';
import { Provider } from 'react-redux';

export const renderTheme = (
  children: React.ReactNode,
  store?: any,
): RenderResult => {
  return render(
    <Provider store={store ?? storeActual}>
      <ThemeProvider theme={theme}>{children}</ThemeProvider>
    </Provider>,
  );
};
