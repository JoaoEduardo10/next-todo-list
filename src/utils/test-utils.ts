import { configureStore } from '@reduxjs/toolkit';
import rootReducer from '../../src/app/store';

export const createTestStore = () => {
  const store = configureStore({
    reducer: rootReducer,
    devTools: false,
  });

  return store;
};
