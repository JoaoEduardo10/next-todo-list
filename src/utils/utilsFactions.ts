import { Tasks } from '../components/task/styles';
import { TTasks } from '../types';

const returnedSubTasksConcluded = (task: TTasks) => {
  const allSubTaskConcluded = task.subTasks.filter((total) => {
    return total.concluded == true;
  });

  return allSubTaskConcluded.length;
};

const returnTasksByStatus = (tasks: TTasks[]) => {
  const tasksConcluidedFilter = tasks.filter((task) => {
    return task.status == 'concluded';
  });

  const tasksPendingFilter = tasks.filter((task) => {
    return task.status == 'pending';
  });

  const tasksProgressFilter = tasks.filter((task) => {
    return task.status == 'progress';
  });

  return { tasksConcluidedFilter, tasksPendingFilter, tasksProgressFilter };
};

export { returnedSubTasksConcluded, returnTasksByStatus };
