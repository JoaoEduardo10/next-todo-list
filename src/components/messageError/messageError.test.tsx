import { screen } from '@testing-library/react';
import { MessageError } from '.';
import { renderTheme } from '../../utils/render-theme';

describe('<MessageError />', () => {
  it('should show the message but the animation of showMessage', () => {
    renderTheme(<MessageError text="Test Error" error={true} />);

    const message = screen.queryByLabelText('Message Error');

    expect(message).toBeInTheDocument();
    expect(message).toHaveStyle({
      animation: 'showMessage 300ms ease-in-out',
    });
  });

  it('should hiden the error message with animation errorMessageNotExist', () => {
    renderTheme(<MessageError text="Test Error" error={false} />);

    const message = screen.queryByLabelText('Message Error');

    expect(message).toBeInTheDocument();
    expect(message).toHaveStyle({
      animation: 'hidenShowMessageError 1s ease-in-out',
    });
  });

  it('should create to Match Snapshot', () => {
    renderTheme(<MessageError text="Test Error" error={false} />);

    const message = screen.queryByLabelText('Message Error');

    expect(message).toMatchSnapshot();
  });
});
