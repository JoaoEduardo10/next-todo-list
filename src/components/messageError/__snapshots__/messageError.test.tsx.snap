// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MessageError /> should create to Match Snapshot 1`] = `
.c0 {
  position: fixed;
  z-index: 3;
  background-color: #ad081b;
  width: 40rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 2rem;
  padding: 1.6rem;
  border-radius: 1.6rem;
  opacity: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  -webkit-animation: hidenShowMessageError 1s ease-in-out;
  animation: hidenShowMessageError 1s ease-in-out;
}

.c0 span {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c0 span::before {
  content: '';
  position: absolute;
  right: -50%;
  width: 0.2rem;
  height: 4rem;
  border-radius: 0.5rem;
  background-color: #d9bf6c;
}

.c0 span svg {
  color: #F4F7FD;
}

@media (max-width:1500px) and (min-width:951px) {
  .c0 {
    top: 10%;
  }
}

@media (max-width:950px) and (min-width:781px) {
  .c0 {
    top: 10%;
    height: 17rem;
    width: 90%;
    font-size: 3.5rem;
  }
}

@media (max-width:780px) {
  .c0 {
    top: 10%;
    width: 81%;
  }
}

<div
  aria-label="Message Error"
  class="c0"
>
   
  <span>
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      stroke-width="0"
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M512 64C264.6 64 64 264.6 64 512s200.6 448 448 448 448-200.6 448-448S759.4 64 512 64zm32 664c0 4.4-3.6 8-8 8h-48c-4.4 0-8-3.6-8-8V456c0-4.4 3.6-8 8-8h48c4.4 0 8 3.6 8 8v272zm-32-344a48.01 48.01 0 0 1 0-96 48.01 48.01 0 0 1 0 96z"
      />
    </svg>
  </span>
   
  <span
    aria-label="Message"
  >
    Test Error
  </span>
</div>
`;
